src/web/src/components/SignUp/Schema/FormSchema.tsx

```diff
@@ -0,0 +1,68 @@
+import * as Yup from 'yup';
+
+import formModels from './FormModel';
+
+const {
+  firstName,
+  lastName,
+  displayName,
+  githubUsername,
+  github,
+  githubOwnership,
+  feeds,
+  blogUrl,
+  blogOwnership,
+} = formModels;
+
+const validateLength = (min: number, max: number) => (val: string | undefined): boolean =>
+  !!val && val.length >= min && val.length <= max;
+
+const validateCheckBox = (val: boolean | undefined) => !!val;
+
+// Each signup step has one validation schema
+export default [
+  // First step has no validation logic
+  Yup.object().shape({}),
+
+  Yup.object().shape({
+    [firstName.name]: Yup.string()
+      .required(`${firstName.requiredErrorMsg}`)
+      .test('len', firstName.invalidErrorMsg, validateLength(2, 16)),
```

Comment:
Why not use the built-in string length assertions? https://github.com/jquense/yup#stringlengthlimit-number--ref-message-string--function-schema

Reply:
Also, where did you come up with 2 and 16?  What if a user's name is longer?  I guarantee we'll have longer values.

Reply:
I would suggest we require the name to exist and nothing about max length.

Reply:
> I would suggest we require the name to exist and nothing about max length.

Thanks, I wasn't sure what the right limits were.

Reply:
Done.